### ISSUE TITLE
home-assistant: cannot set `hvac_mode` to get the value `"off"` in the configuration

### ISSUE THREAD

**Issue by itepastra** [NixOS/nixpkgs#354839]
### Describe the bug
When I try to set the key `hvac_mode` to the value `"off"` the value gets converted to `False` which does not work. 
Adding more quotes does not help since nix will escape the inside quotes instead.
There should be a way to create the line `hvac_mode: "off"` in the yaml

### Steps To Reproduce
1. in home assistant config for an automation action, add `hvac_mode = "off";`
2. rebuild the configuration.yaml
3. configuration.yaml contains `False` instead of `"off"`

### Expected behavior
The configuration file contains `"off"`. Or there is a way to make it

### Screenshots
![image](https://github.com/user-attachments/assets/97991587-446b-4efe-87ee-288e4aa4202b)
![image](https://github.com/user-attachments/assets/fe9a90ab-f444-409a-915d-b4acbf440489)

### Metadata

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by jackwilsdon** [comment]
It looks like this may be fixed by #348098 - it upgrades `remarshal` past 0.18 where it changed to using YAML 1.2, which no longer treats "on" / "off" as booleans.

**Comment by itepastra** [comment]
I'm on the master branch, commit fcfcc8e

**Comment by jackwilsdon** [comment]
Hmm that's odd. I can't reproduce this on 5763214382961a36a54f259bbf80485664b9664b:

```shell
$ cat home-assistant.nix
{
  services.home-assistant = {
    enable = true;
    config.automations = [ { action = [ { hvac_mode = "off"; } ]; } ];
  };
}
$ nix build .#nixosConfigurations.my-system.config.environment.etc."\"home-assistant/configuration.yaml\"".source
$ cat result 
automations:
- action:
  - hvac_mode: off
homeassistant:
  time_zone: Europe/London
http:
  server_host:
  - 0.0.0.0
  - '::'
  server_port: 8123
lovelace:
  mode: storage
  resources: []
```

**Comment by JManch** [comment]
This should be fixed by #354687
